### PR TITLE
Fixes for subquery indexing/correlation tracking

### PR DESCRIPTION
**Pull request by max-hoffman**
I missed a place where we were using getField indexes to do a correlation check. This adds the changes necessary to replace that check with one that uses a subquery's tracked correlation column set. This also adds an expression id column to GetField expression that preserves the id tracking information between iterative passes of the analyzer. It would probably be preferable to avoid all cases where we unnecessarily re-run rules, but this is more near at hand.

### CODE REVIEW

File: sql/plan/scope.go [hunk 1]
```diff
@@ -92,9 +93,13 @@ func (s *Scope) NewScope(node sql.Node) *Scope {
 
 // NewScopeFromSubqueryExpression returns a new subscope created from a subquery expression contained by the specified
 // node.
-func (s *Scope) NewScopeFromSubqueryExpression(node sql.Node) *Scope {
+func (s *Scope) NewScopeFromSubqueryExpression(node sql.Node, corr sql.ColSet) *Scope {
```
Comment (fulghum) [inline sql/plan/scope.go:96]: Would be helpful to update the method doc to mention the `corr` parameter and explain what it means here. It's probably painfully obvious to you of course, but a little documentation there would help other people from having to reverse engineer it when they see this code for the first time.

File: sql/planbuilder/scope.go [hunk 2]
```diff
@@ -127,6 +128,7 @@ func (s *scope) resolveColumn(table, col string, checkParent bool) (scopeColumn,
 	if !foundCand {
 		return scopeColumn{}, false
 	}
+	//s.parent.nearestSubquery().addOutOfScope(c.id)
```
Comment (fulghum) [inline sql/planbuilder/scope.go:131]: Might as well pull out dead code
```suggestion
```

**Review by fulghum — APPROVED** [review]
Nice! Awesome to see all those `IndexedTableAccess` nodes come back 😄

File: sql/rowexec/ddl_iters.go [hunk 3]
```diff
@@ -1460,7 +1460,7 @@ func addColumnToSchema(schema sql.Schema, column *sql.Column, order *sql.ColumnO
 						if idx < 0 {
 							return nil, transform.SameTree, sql.ErrTableColumnNotFound.New(schema[0].Source, s.Name())
 						}
-						return s.WithIndex(idx), transform.NewTree, nil
+						return expression.NewGetFieldWithTable(idx, s.Type(), s.Table(), s.Name(), s.IsNullable()), transform.NewTree, nil
```
[the remark this answers is not in the document]
Comment (max-hoffman) [reply sql/rowexec/ddl_iters.go:1463]: Strictly a testing thing. Column default parsing will give the first expression index=1 and id=1. The index fixup changes index=0, but leaves id=1. But the testing expected index=0 and id=0, so I just made it reset the id also.

File: sql/plan/scope.go (hunk 1 above)
Comment (max-hoffman) [reply sql/plan/scope.go:96]: added some attribute docs for what it means, and doc comment here to indicate why it needs to be explicit for subquery expressions

**Pull request merged**